src/PremiumLib.cpp

```diff
@@ -0,0 +1,429 @@
+/**
+    This plugin can be used for common player customizations
+ */
+
+#include "AccountMgr.h"
+#include "ScriptMgr.h"
+#include "Player.h"
+#include "Config.h"
+#include "Chat.h"
+#include "Language.h"
+#include "CharacterDatabase.h"
+#include "LoginDatabase.h"
+#include "DatabaseWorkerPool.h"
+#include "MySQLConnection.h"
+#include "PremiumLib.h"
+
+/* Premium character interaction.
+     * These functions will allow modules to access the new generic
+     * premium tables designed to help modules that interact with
+     * players and characters without the need to create other tables.
+*/
+
+PremiumLibData PremiumLibManager::GetAccountPremiumLevel(uint64 accountID)
+{
+    QueryResult result = LoginDatabase.PQuery("SELECT premium_level, expiration_date, FROM_UNIXTIME(expiration_date) FROM premium_account WHERE account_id = %u", accountID);
+
+    if (!result)
+        return PremiumLibData();
+
+    PremiumLibData data;
+    data.premiumLevel = (*result)[0].GetInt8();
+    data.expirationDateUnixtime = (*result)[1].GetInt32();
+    data.expirationDate = (*result)[2].GetCString();
+    return data;
+}
+
+PremiumLibData PremiumLibManager::GetCharacterPremiumLevel(uint64 guid)
+{
+    QueryResult result = CharacterDatabase.PQuery("SELECT premium_level, expiration_date, FROM_UNIXTIME(expiration_date) FROM premium_character WHERE character_id = %u", GUID_LOPART(guid));
+
+    if (!result)
+        return PremiumLibData();
+
+    PremiumLibData data;
+    data.premiumLevel = (*result)[0].GetInt8();
+    data.expirationDateUnixtime = (*result)[1].GetInt32();
+    data.expirationDate = (*result)[2].GetCString();
+    return data;
+}
+
+bool PremiumLibManager::CreateAccountPremiumLevel(uint64 accountID, int8 premiumLevel)
```

Comment:
I suggest to change "Create" to "Set", even if it effectively creates a new row that didn't exist in the DB. But it's more like setting a... setting :P